src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -151,3 +156,66 @@ def signature_hash_algorithm(self):
             raise UnsupportedAlgorithm(
                 "Signature algorithm OID:{0} not recognized".format(oid)
             )
+
+    @property
+    def extensions(self):
+        # TODO: remove this when we finish landing support for extensions.
+        warnings.warn(
+            "Extension support is not fully implemented. Unsupported "
+            "extensions are IGNORED at this time."
+        )
```

Comment:
How complex do you think handling those extensions is?

My goal is to find a strategy that doesn't leave us in a state where doing
a release is totally blocked.

On Fri, Mar 27, 2015 at 10:39 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/x509.py
> <https://github.com/pyca/cryptography/pull/1769#discussion_r27341675>:
>
> > @@ -151,3 +156,66 @@ def signature_hash_algorithm(self):
> >              raise UnsupportedAlgorithm(
> >                  "Signature algorithm OID:{0} not recognized".format(oid)
> >              )
> > +
> > +    @property
> > +    def extensions(self):
> > +        # TODO: remove this when we finish landing support for extensions.
> > +        warnings.warn(
> > +            "Extension support is not fully implemented. Unsupported "
> > +            "extensions are IGNORED at this time."
> > +        )
>
> The CA false test (which uses the cryptography.io cert itself) and the
> test without basic constraints (which uses PKITS data from NIST) both
> contain critical key usage extensions that would cause failures. Once we
> land support for that extension we'd be okay though.
>
> Alternately we could replace those vectors with more specific ones, but
> I'll need to submit a new vector PR for that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1769/files#r27341675>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084